txstratum/utils.py

```diff
@@ -404,3 +404,23 @@ def _exception_handler(loop: "AbstractEventLoop", context: Dict[str, Any]) -> No
         logger.exception(message, **extra, exc_info=exc_info)
 
     loop.set_exception_handler(_exception_handler)
+
+
+def is_version_gte(version: str, base_version: str) -> bool:
+    """Compare two versions in the format X.Y.Z.
+
+    Validates if version is greater than or equal to base_version
+    """
+    # First we transform the versions from strings in the format X.Y.Z
+    # into tuples of integers
+
+    try:
+        version_int_tuple = tuple(map(int, version.split(".")))
+        base_version_int_tuple = tuple(map(int, base_version.split(".")))
+    except ValueError:
+        # version strings are not in the expected format
+        return False
+
+    assert len(version_int_tuple) == len(base_version_int_tuple)
```

Comment:
I feel we should have an if and raise an exception (instead of using an assert).